### DISCUSSION TITLE
Openshift (kubernetes) PV - no write permission

### DISCUSSION THREAD

**Discussion by maciejlesny** [ceph/ceph-csi#3334] [General]
Hi,
I've been looking for possibility adding Ceph to Openshift 4.9 as persistent volume for containers. Everything works great, except mount point has "drwxr-xr-x" (755) permissions. Openshift doesn't allow root user inside containers. I cannot chown/chmod mounted point.
I've tried Ceph RBD and CephFS (filesystems).

Is there any way to set default permissions to 777?

**Comment by Madhu-1** [comment]
setting 777 on both cephfs and rbd mounts are removed in cephcsi, Have you considered setting `SecurityContext:` in both pod spec and container spec? https://github.com/ceph/ceph-csi/blob/3ddec80346f6fe6ae6da8c19adfdb92e49e41fa8/e2e/utils.go#L515-L537

**Comment by maciejlesny** [comment]
I'm looking for persistent volume solution which will work without modification deployments/pods. So using openshift with ceph-csi is not possible, am I right?

**Comment by Madhu-1** [comment]
> I'm looking for persistent volume solution which will work without modification deployments/pods. So using openshift with ceph-csi is not possible, am I right?

AFAIK Without modifying pod/deployment spec not possible
